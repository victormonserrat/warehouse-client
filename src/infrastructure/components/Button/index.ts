import styled from 'styled-components'

const Button = styled.button`
  background-color: ${({ theme }) => theme.colors.blueLight};
  border-radius: 2.5rem;
  color: ${({ theme }) => theme.colors.white};
  font-family: ${({ theme }) => theme.fonts.primary};
  font-size: ${({ theme }) => theme.sizes.small};
  font-weight: bold;
  padding: 1.25rem 2.5rem;
  width: 100%;
  transition: background-color 0.3s ease;

  :hover,
  :focus {
    background-color: ${({ theme }) => theme.colors.blue};
  }

  :active {
    background-color: ${({ theme }) => theme.colors.blueDark};
  }

  :disabled {
    background-color: ${({ theme }) => theme.colors.white};
    box-shadow: inset 0 0 0 0.0625rem ${({ theme }) => theme.colors.greyLight};
    color: ${({ theme }) => theme.colors.greyLight};
  }
`
export default Button
