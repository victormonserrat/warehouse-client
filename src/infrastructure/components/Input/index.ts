import styled from 'styled-components'

const Input = styled.input`
  box-shadow: 0 0.0625rem 0 ${({ theme }) => theme.colors.grey};
  color: ${({ theme }) => theme.colors.black};
  font-family: ${({ theme }) => theme.fonts.primary};
  font-size: ${({ theme }) => theme.sizes.medium};
  padding: 0.25rem 0;
  width: 100%;

  :focus {
    box-shadow: 0 0.125rem 0 ${({ theme }) => theme.colors.blue};
    color: ${({ theme }) => theme.colors.blue};
  }

  &[aria-invalid='true'] {
    box-shadow: 0 0.125rem 0 ${({ theme }) => theme.colors.red};
    color: ${({ theme }) => theme.colors.red};
  }

  :disabled {
    box-shadow: 0 0.0625rem 0 ${({ theme }) => theme.colors.greyLight};
    color: ${({ theme }) => theme.colors.greyLight};
  }
`

export default Input
