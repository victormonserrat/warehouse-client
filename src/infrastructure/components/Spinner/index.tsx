import { memo, SVGAttributes } from 'react'

import { Circle, Component } from './styles'

const Spinner = ({ ...props }: SVGAttributes<SVGSVGElement>) => (
  <Component aria-label="Loading" viewBox="0 0 50 50" {...props}>
    <Circle cx="25" cy="25" r="20"></Circle>
  </Component>
)

export default memo(Spinner)
