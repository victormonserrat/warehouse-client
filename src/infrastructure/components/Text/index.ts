import styled, { css } from 'styled-components'

export const BodyLargeStyles = css`
  color: ${({ theme }) => theme.colors.black};
  font-family: ${({ theme }) => theme.fonts.primary};
  font-size: ${({ theme }) => theme.sizes.large};
  font-weight: bold;
  text-transform: uppercase;
`

export const BodyLarge = styled.div`
  ${BodyLargeStyles}
`

export const BodySmallStyles = css`
  color: ${({ theme }) => theme.colors.greyDark};
  font-family: ${({ theme }) => theme.fonts.primary};
  font-size: ${({ theme }) => theme.sizes.small};
`

export const BodySmall = styled.div`
  ${BodySmallStyles}
`

export const H1Styles = css`
  color: ${({ theme }) => theme.colors.black};
  font-family: ${({ theme }) => theme.fonts.primary};
  font-size: ${({ theme }) => theme.sizes.xl};
  font-weight: bold;
`

export const H1 = styled.h1`
  ${H1Styles}
`
