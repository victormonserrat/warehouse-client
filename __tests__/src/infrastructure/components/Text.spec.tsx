import { BodyLarge, BodySmall, H1 } from '~/infrastructure/components/Text'
import { render } from '~/tests/utils'

describe('Body Large', () => {
  it('renders correctly', () => {
    const { container } = render(<BodyLarge>Jest</BodyLarge>)

    expect(container).toMatchSnapshot('Regular')
  })
})

describe('Body Small', () => {
  it('renders correctly', () => {
    const { container } = render(<BodySmall>Jest</BodySmall>)

    expect(container).toMatchSnapshot('Regular')
  })
})

describe('Heading 1', () => {
  it('renders correctly', () => {
    const { container } = render(<H1>Jest</H1>)

    expect(container).toMatchSnapshot('Regular')
  })
})
