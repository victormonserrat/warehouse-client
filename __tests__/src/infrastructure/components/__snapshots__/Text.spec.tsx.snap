// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body Large renders correctly: Regular 1`] = `
<div>
  <div
    class="sc-bczRLJ dcfFMD"
  >
    Jest
  </div>
</div>
`;

exports[`Body Small renders correctly: Regular 1`] = `
<div>
  <div
    class="sc-gsnTZi gUgIWf"
  >
    Jest
  </div>
</div>
`;

exports[`Heading 1 renders correctly: Regular 1`] = `
<div>
  <h1
    class="sc-dkzDqf SuHUx"
  >
    Jest
  </h1>
</div>
`;
